### ISSUE TITLE
server side validation in asp dotNet core

### ISSUE THREAD

**Issue by soraraso42** [soraraso42/learning-journal#46]
---
#### ** using partial view on front-end ** ####
basically just dynamically injecting ONLY the necessary JS, JQuery and HTML scripts

Using Partial View:

html

@section Scripts {
    <partial name="_ValidationScriptsPartial" />
}

This line injects the same jQuery validation libraries through the _ValidationScriptsPartial.cshtml file, which is the recommended and more modular way.

When you include the partial view (`<partial name="_ValidationScriptsPartial" />`), the Razor Pages engine doesn't render unnecessary content—it only injects the specific JavaScript libraries and validation logic needed for client-side validation. Here’s a breakdown:

1. **Selective Loading**:
   - The partial view only contains references to the **necessary scripts** (like jQuery and the validation plugins).
   - This is efficient because it keeps your main page (`Index.cshtml`) clean and avoids unnecessary repetition of script references across multiple pages.

2. **Razor Pages + jQuery Validation**:
   - Razor Pages automatically generate **data attributes** (e.g., `data-val`, `data-val-required`) based on the validation attributes in your `InputModel` (like `[Required]`, `[EmailAddress]`).
   - The **unobtrusive jQuery validation** script reads these attributes and handles the actual validation process without you needing to write JavaScript manually.
   - The feedback (like displaying an error message when a field is empty) happens **instantly on the client side**—this avoids submitting the form to the server when there are validation issues.

3. **Validation Feedback**:
   - The validation errors are displayed **dynamically** in the form based on the rules you've set in your `InputModel`.
   - Razor Pages handle showing the error messages (usually below the relevant input fields) using the `asp-validation-for` tag helper, which works in conjunction with the validation logic provided by the partial view scripts.

### Why Use Partial View for Scripts?
By keeping the validation scripts in a partial view:
- You **avoid duplicating** the script references in every page.
- It makes it **modular and reusable**. If you need to update or change the validation libraries, you only need to change the partial view, and it applies to all pages using it.
- It allows the **Razor Pages engine** to handle just the required functionality for client-side validation based on the model’s validation attributes.

### In Summary:
The partial view for validation scripts ensures that Razor Pages injects just the necessary JavaScript for validating your form on the client side, making your business logic easier to maintain and more efficient, with real-time validation feedback for users.
